src/java/org/apache/sqoop/mapreduce/MergeMapperBase.java

```diff
@@ -76,9 +76,10 @@ protected void processRecord(SqoopRecord r, Context c)
     }
     Object keyObj = null;
     if (keyColName.contains(",")) {
+        String connectStr = new String(new byte[]{1});
         StringBuilder keyFieldsSb = new StringBuilder();
         for (String str : keyColName.split(",")) {
-            keyFieldsSb.append("+").append(fieldMap.get(str).toString());
+            keyFieldsSb.append(connectStr).append(fieldMap.get(str).toString());
```

Comment:
for example one table has two column, a and b

Field a | Field b
------------ | -------------
a+ | b
a | +b

when use "+" to connect two field, two record will has same keyObj.
To avoid this i use a String contains one byte.